src/integrations.py

```diff
@@ -0,0 +1,233 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
```

Comment:
i'm wondering how much this file can grow, the more integrations we have the more this will kinda resemble what we are trying to split, which is `charm.py`. Is there a plan to structure integrations too?

Reply:
I don't think they are the same thing. The target is to offload unnecessary logic in `charm.py` to other components so that it holds more clear responsibility.

For `integrations.py`, its responsibility is already clear although it could grow bigger. But this does not affect the component's responsibility.